In [1]:
# setup

import pandas as pd 
import numpy as np

# This project is about the popularity of board games, and what makes a board game successful.

One of my main hobbies is board games. Board games have evolved significantly from the classics, such as _Monopoly_, _Scabble_. and _The Game of Life_. The modern boardgame is incredibly diverse, with different mechanics, from card drafting, worker-placing and resource management. 

One of the major board game sites, <a href=https://boardgamegeek.com/>_BoardGameGeek_</a>, is a platform for the community to document releases, provide ratings and reviews, and discuss the intracies of the hobby.
This project will be using this dataset: <a href=https://www.kaggle.com/datasets/jvanelteren/boardgamegeek-reviews>BoardGameGeek Reviews</a>
I'll be asking questions and searching for answers throughout the project.

## Data import and cleaning.

The dataset for this project is split into 5 CSV files. 
<ul>
    <li>A list of 15 Million reviews, collected in 2020</li>
    <li>A list of 19 Million reviews, collected in 2022</li>
    <li>A list of Board games with over 30 reviews, collected in 2020</li>
    <li>A list of Board games with over 30 reviews, collected in 2022</li>
    <li>A list of board games with detailed information.</li>
</ul>

In [4]:
data = "./data"
df_20 = pd.read_csv(f"{data}/2020-08-19.csv")
df_20.head()

Unnamed: 0     ID         Name  Year  Rank  Average  Bayes average  \
0          90  30549     Pandemic  2008    91     7.62          7.518   
1         172    822  Carcassonne  2000   173     7.42          7.311   
2         380     13        Catan  1995   381     7.16          7.001   
3          49  68448    7 Wonders  2010    50     7.76          7.662   
4          87  36218     Dominion  2008    88     7.63          7.521   

   Users rated                         URL  \
0        96186   /boardgame/30549/pandemic   
1        96181  /boardgame/822/carcassonne   
2        96171         /boardgame/13/catan   
3        79830  /boardgame/68448/7-wonders   
4        74933   /boardgame/36218/dominion   

                                           Thumbnail  
0  https://cf.geekdo-images.com/micro/img/0m3-oqB...  
1  https://cf.geekdo-images.com/micro/img/z0tTaij...  
2  https://cf.geekdo-images.com/micro/img/e0y6Bog...  
3  https://cf.geekdo-images.com/micro/img/h-Ejv31...  
4  https://cf.geekdo-images.com/micro/img/VYp2s2f...

In [5]:
df_20.set_index('ID', inplace = True)

In [6]:
display(df_20[df_20.duplicated()])
df_20.isna().value_counts()

Empty DataFrame
Columns: [Unnamed: 0, Name, Year, Rank, Average, Bayes average, Users rated, URL, Thumbnail]
Index: []

Unnamed: 0  Name   Year   Rank   Average  Bayes average  Users rated  URL    Thumbnail
False       False  False  False  False    False          False        False  False        19317
                                                                             True            13
Name: count, dtype: int64

In [7]:
# The data is very clean, with no missing values apart from 13 thumbnails. I'll drop this column, and other inimportant columns..
to_drop = ["Thumbnail", "URL", "Unnamed: 0"]
df_20.drop(to_drop, axis = 1, inplace = True)

In [8]:
df_22 = pd.read_csv(f"{data}/2022-01-08.csv")
df_22.head()

Unnamed: 0     ID         Name  Year  Rank  Average  Bayes average  \
0         105  30549     Pandemic  2008   106     7.59          7.487   
1         189    822  Carcassonne  2000   190     7.42          7.309   
2         428     13        Catan  1995   429     7.14          6.970   
3          72  68448    7 Wonders  2010    73     7.74          7.634   
4         103  36218     Dominion  2008   104     7.61          7.499   

   Users rated                         URL  \
0       108975   /boardgame/30549/pandemic   
1       108738  /boardgame/822/carcassonne   
2       108024         /boardgame/13/catan   
3        89982  /boardgame/68448/7-wonders   
4        81561   /boardgame/36218/dominion   

                                           Thumbnail  
0  https://cf.geekdo-images.com/S3ybV1LAp-8SnHIXL...  
1  https://cf.geekdo-images.com/okM0dq_bEXnbyQTOv...  
2  https://cf.geekdo-images.com/W3Bsga_uLP9kO91gZ...  
3  https://cf.geekdo-images.com/RvFVTEpnbb4NM7k0I...  
4  https://cf.geekdo-images.com/j6iQpZ4XkemZP07HN...

In [9]:
df_22.set_index('ID', inplace = True)

In [10]:
display(df_22[df_22.duplicated()])
df_22.isna().value_counts()

Empty DataFrame
Columns: [Unnamed: 0, Name, Year, Rank, Average, Bayes average, Users rated, URL, Thumbnail]
Index: []

Unnamed: 0  Name   Year   Rank   Average  Bayes average  Users rated  URL    Thumbnail
False       False  False  False  False    False          False        False  False        21825
                                                                             True             6
Name: count, dtype: int64

In [11]:
# The data is very clean, with no missing values apart from 6 thumbnails. I'll drop this column, and other inimportant columns..
df_22.drop(to_drop, axis = 1, inplace = True)

With just the data from the 2 lists of games, there's a few questions I'd like to ask:
<ol>
    <li>What are the highest rated games in both years?</li>
    <li>What is the average user rating in both years? Do users think the quality of games has gone up or down?</li>
    <li>How many more board games are there in 2022, compared to 2020?</li>
    <li>Which games had the highest change in rank?</li>
</ol>


### 1. What are the highest rated games in both years?

In [13]:
top20_2020 = df_20.sort_values(by='Rank').head(20)
top20_2020

Name  Year  Rank  Average  \
ID                                                                           
174430                                     Gloomhaven  2017     1     8.82   
161936                      Pandemic Legacy: Season 1  2015     2     8.62   
224517                              Brass: Birmingham  2018     3     8.64   
167791                              Terraforming Mars  2016     4     8.43   
233078             Twilight Imperium (Fourth Edition)  2017     5     8.69   
182028  Through the Ages: A New Story of Civilization  2015     6     8.45   
220308                                   Gaia Project  2017     7     8.50   
187645                           Star Wars: Rebellion  2016     8     8.42   
12333                               Twilight Struggle  2005     9     8.30   
193738                            Great Western Trail  2016    10     8.29   
169786                                         Scythe  2016    11     8.25   
115746                War of the Ring: Second Edition  2012    12     8.47   
162886                                  Spirit Island  2017    13     8.32   
84876                         The Castles of Burgundy  2011    14     8.13   
120677                                  Terra Mystica  2012    15     8.16   
173346                                 7 Wonders Duel  2015    16     8.10   
124361                                      Concordia  2013    17     8.12   
28720                               Brass: Lancashire  2007    18     8.16   
205637                   Arkham Horror: The Card Game  2016    19     8.19   
266192                                       Wingspan  2019    20     8.11   

        Bayes average  Users rated  
ID                                  
174430          8.569        37053  
161936          8.470        38345  
224517          8.305        14531  
167791          8.282        57319  
233078          8.213        11466  
182028          8.211        21035  
220308          8.162        14270  
187645          8.160        21089  
12333           8.148        38869  
193738          8.109        26372  
169786          8.093        52485  
115746          8.080        12436  
162886          8.056        20726  
84876           8.013        43508  
120677          8.006        37267  
173346          7.975        53475  
124361          7.958        25698  
28720           7.949        18140  
205637          7.929        25045  
266192          7.925        32712

In [14]:
top20_2022 = df_22.sort_values(by='Rank').head(20)
top20_2022

Name  Year  Rank  Average  \
ID                                                                           
174430                                     Gloomhaven  2017     1     8.74   
161936                      Pandemic Legacy: Season 1  2015     2     8.59   
224517                              Brass: Birmingham  2018     3     8.66   
167791                              Terraforming Mars  2016     4     8.42   
233078              Twilight Imperium: Fourth Edition  2017     5     8.68   
291457                   Gloomhaven: Jaws of the Lion  2020     6     8.68   
220308                                   Gaia Project  2017     7     8.47   
187645                           Star Wars: Rebellion  2016     8     8.42   
182028  Through the Ages: A New Story of Civilization  2015     9     8.38   
115746                War of the Ring: Second Edition  2012    10     8.51   
162886                                  Spirit Island  2017    11     8.36   
193738                            Great Western Trail  2016    12     8.29   
12333                               Twilight Struggle  2005    13     8.28   
169786                                         Scythe  2016    14     8.22   
84876                         The Castles of Burgundy  2011    15     8.13   
173346                                 7 Wonders Duel  2015    16     8.11   
167355                                        Nemesis  2018    17     8.39   
120677                                  Terra Mystica  2012    18     8.12   
124361                                      Concordia  2013    19     8.12   
28720                               Brass: Lancashire  2007    20     8.18   

        Bayes average  Users rated  
ID                                  
174430          8.511        47827  
161936          8.442        45041  
224517          8.418        25484  
167791          8.274        74216  
233078          8.262        16025  
291457          8.260        15918  
220308          8.175        19169  
187645          8.172        25586  
182028          8.150        25605  
115746          8.136        15498  
162886          8.133        32199  
193738          8.123        32960  
12333           8.113        43052  
169786          8.064        64569  
84876           8.008        50549  
173346          7.984        69472  
167355          7.982        17688  
120677          7.966        41806  
124361          7.966        31519  
28720           7.963        20915

I've chosen an arbitary value to look at the top games in. But how many games stayed in the top 20? and which ones?

In [16]:
stayed_in_top20_ids = []
for id in top20_2020.index:
    if id in top20_2022.index:
        stayed_in_top20_ids.append(id)
stayed_in_top20 = top20_2022.loc[stayed_in_top20_ids]
stayed_in_top20


#stayedString = ""
#for name in stayed_in_top20['Name'].values:
#    stayedString = stayedString + f"{name}\n" 
#print(f"{stayed_in_top20.shape[0]} games stayed in the top 20. The ones that stayed were: \n{stayedString}")

Name  Year  Rank  Average  \
ID                                                                           
174430                                     Gloomhaven  2017     1     8.74   
161936                      Pandemic Legacy: Season 1  2015     2     8.59   
224517                              Brass: Birmingham  2018     3     8.66   
167791                              Terraforming Mars  2016     4     8.42   
233078              Twilight Imperium: Fourth Edition  2017     5     8.68   
182028  Through the Ages: A New Story of Civilization  2015     9     8.38   
220308                                   Gaia Project  2017     7     8.47   
187645                           Star Wars: Rebellion  2016     8     8.42   
12333                               Twilight Struggle  2005    13     8.28   
193738                            Great Western Trail  2016    12     8.29   
169786                                         Scythe  2016    14     8.22   
115746                War of the Ring: Second Edition  2012    10     8.51   
162886                                  Spirit Island  2017    11     8.36   
84876                         The Castles of Burgundy  2011    15     8.13   
120677                                  Terra Mystica  2012    18     8.12   
173346                                 7 Wonders Duel  2015    16     8.11   
124361                                      Concordia  2013    19     8.12   
28720                               Brass: Lancashire  2007    20     8.18   

        Bayes average  Users rated  
ID                                  
174430          8.511        47827  
161936          8.442        45041  
224517          8.418        25484  
167791          8.274        74216  
233078          8.262        16025  
182028          8.150        25605  
220308          8.175        19169  
187645          8.172        25586  
12333           8.113        43052  
193738          8.123        32960  
169786          8.064        64569  
115746          8.136        15498  
162886          8.133        32199  
84876           8.008        50549  
120677          7.966        41806  
173346          7.984        69472  
124361          7.966        31519  
28720           7.963        20915

In [17]:
left_top20 = top20_2020[~top20_2020.index.isin(top20_2022.index)]
entered_top20 = top20_2022[~top20_2022.index.isin(top20_2020.index)]
print("These games left the top 20:")
display(left_top20)
print("These games entered the top 20:")
display(entered_top20)

These games left the top 20:


Name  Year  Rank  Average  Bayes average  \
ID                                                                         
205637  Arkham Horror: The Card Game  2016    19     8.19          7.929   
266192                      Wingspan  2019    20     8.11          7.925   

        Users rated  
ID                   
205637        25045  
266192        32712

These games entered the top 20:


Name  Year  Rank  Average  Bayes average  \
ID                                                                         
291457  Gloomhaven: Jaws of the Lion  2020     6     8.68          8.260   
167355                       Nemesis  2018    17     8.39          7.982   

        Users rated  
ID                   
291457        15918  
167355        17688

In [18]:
#What is the average score for both years?
avg_2022 = df_22['Average'].mean()
avg_2020 = df_20['Average'].mean()
print(f"The average score in 2020 was: {avg_2020} \nThe average score in 2022 was: {avg_2022}")

# How many games beat the average?
beat_avg_22 = (df_22['Average'] > avg_2022).sum()
beat_avg_20 = (df_20['Average'] > avg_2020).sum()
print(f"In 2020, {beat_avg_20} games beat the average. In 2022, {beat_avg_22} games beat the average.")

The average score in 2020 was: 6.392255561303674 
The average score in 2022 was: 6.416960285832074
In 2020, 9880 games beat the average. In 2022, 11240 games beat the average.


In [19]:
# How about the games that lost/gained the most ranks?

# 3 States a game can be in
# In both lists
# in first list
# in second list 

df_22_rank_change = df_22.copy()

def rank_change(id):
    return df_20.loc[id]['Rank'] - df_22.loc[id]['Rank']

ids_20 = df_20.index
ids_22 = df_22.index
## check if a game is in both and calculate the change:
for id in ids_22:
    if id in ids_20:
        change = rank_change(id)
        df_22_rank_change.loc[id,'Ranked change'] = change


df_22_rank_change.sample(3)


Name  Year   Rank  Average  Bayes average  Users rated  \
ID                                                                          
145586           sQuizz  2013  13990     6.35          5.523           48   
113995             Jask  2010  17062     6.08          5.506           32   
3086    The Omega Virus  1992   3829     6.41          5.803         1076   

        Ranked change  
ID                     
145586        -1323.0  
113995            NaN  
3086           -425.0

In [20]:
print("The games that gained the most ranks were:")
display(df_22_rank_change.sort_values(by= 'Ranked change', ascending=False).head(3))
print("The games that lost the most ranks were:")
display(df_22_rank_change.sort_values(by= 'Ranked change', ascending=False, na_position ='first' ).tail(3))

The games that gained the most ranks were:


Name  Year  Rank  Average  Bayes average  Users rated  \
ID                                                                           
273703        Altar Quest  2020  1500     7.92          6.291          972   
281515  Company of Heroes  2021  2624     8.83          5.984          401   
295905        Cosmic Frog  2020  2514     7.53          6.005          580   

        Ranked change  
ID                     
273703        11349.0  
281515        10259.0  
295905        10234.0

The games that lost the most ranks were:


Name  Year   Rank  Average  \
ID                                                                        
236639                           Flaming Pyramids  2018  20185     5.41   
212027                               Balloon Pop!  2017  21440     5.05   
240463  Tournament Fishing: The Deckbuilding Game  2017  21633     5.02   

        Bayes average  Users rated  Ranked change  
ID                                                 
236639          5.479          283        -8267.0  
212027          5.426          314        -8788.0  
240463          5.374          352       -14758.0

In [21]:
#Actually not that interesting until we find out how many new games were added to the catalogue, and how many were taken away...
# Checks in first list/in second list condition
df_removed = df_20[~df_20.index.isin(df_22.index)]
df_added = df_22[~df_22.index.isin(df_20.index)]
removed_no = df_removed.shape[0]
added_no = df_added.shape[0]
print(f"BGG removed {removed_no} games, and added {added_no} games.")

BGG removed 48 games, and added 2549 games.


In [22]:
# What year were the most games released?
years_srs = df_22['Year'].value_counts()
years_srs.head(1)

#How many years have the fewest games?
years_srs[years_srs == years_srs.min()].count()

60

Whilst this information is nice, it's not that useful from an analysis standpoint. What makes these game so good? What kind of game do BGG users like?

Let's start by getting more information on the games.

In [24]:
# Got a Dtype warning on first run, specifying dType of columns 29:Board Game Rank, and 49: Children's game rank
dtypes = { "Board Game Rank" : str,
           "Children's Game Rank": str
         }
df_game_details = pd.read_csv(f"{data}/games_detailed_info.csv", dtype = dtypes)
df_game_details.head()
df_game_details.set_index('id', inplace = True)

In [25]:
df_game_details.info()

<class 'pandas.core.frame.DataFrame'>
Index: 21631 entries, 30549 to 165946
Data columns (total 55 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Unnamed: 0                     21631 non-null  int64  
 1   type                           21631 non-null  object 
 2   thumbnail                      21616 non-null  object 
 3   image                          21616 non-null  object 
 4   primary                        21631 non-null  object 
 5   alternate                      8850 non-null   object 
 6   description                    21630 non-null  object 
 7   yearpublished                  21631 non-null  int64  
 8   minplayers                     21631 non-null  int64  
 9   maxplayers                     21631 non-null  int64  
 10  suggested_num_players          21631 non-null  object 
 11  suggested_playerage            21520 non-null  object 
 12  suggested_language_dependence  21559 non-null 

This dataset is an expanded verion of the 2022 dataset that we used to compare to the 2020 dataset, including many of the same columns. 
There seems to be only 1 game for columns 48-54. I thought that was strange, so it's time to look into it.

In [27]:
display(df_game_details[df_game_details['RPG Item Rank'].notna()].index,
df_game_details[df_game_details['Accessory Rank'].notna()].index,
df_game_details[df_game_details['Video Game Rank'].notna()].index,
df_game_details[df_game_details['Amiga Rank'].notna()].index,
df_game_details[df_game_details['Commodore 64 Rank'].notna()].index,
df_game_details[df_game_details['Arcade Rank'].notna()].index,
df_game_details[df_game_details['Atari ST Rank'].notna()].index)

Index([234669], dtype='int64', name='id')

Index([5985], dtype='int64', name='id')

Index([91313], dtype='int64', name='id')

Index([91313], dtype='int64', name='id')

Index([91313], dtype='int64', name='id')

Index([91313], dtype='int64', name='id')

Index([91313], dtype='int64', name='id')

There are 3 entries with strange ranks: `234669, 5985, 91313`. What are they? 

In [29]:
df_game_details[df_game_details['RPG Item Rank'].notna() | df_game_details['Accessory Rank'].notna() | df_game_details['Video Game Rank'].notna()][['type','primary','Board Game Rank']]

type                       primary Board Game Rank
id                                                             
234669  boardgame          Legacy of Dragonholt            1019
5985    boardgame  Miscellaneous Game Accessory            5423
91313   boardgame                         Ataxx           15599

`Miscellaneous Game Accessory` doesn't seem like a game - so might be worth dropping from the dataframe.
`Ataxx` is a videogame that has a ruleset that can be re-created on a reversi board, but was never sold as a boardgame by itself.
`Legacy of Dragonholt` is a tabletop RPG, so this entry will be staying.

There are more columns that I'm interested in, and it will probably be worth taking a closer look:

In [31]:
display(df_game_details[['primary','Board Game Rank','usersrated','average','bayesaverage','stddev','median']].sort_values(by='bayesaverage', ascending = False).head(5))
display(df_game_details[['primary','Board Game Rank','usersrated','average','bayesaverage','stddev','median']].sort_values(by='average', ascending = False).head(5))

primary Board Game Rank  usersrated  \
id                                                                      
174430                         Gloomhaven               1       47853   
161936          Pandemic Legacy: Season 1               2       45064   
224517                  Brass: Birmingham               3       25515   
167791                  Terraforming Mars               4       74269   
233078  Twilight Imperium: Fourth Edition               5       16036   

        average  bayesaverage   stddev  median  
id                                              
174430  8.74356       8.51055  1.64190       0  
161936  8.59350       8.44208  1.56316       0  
224517  8.66343       8.41823  1.24216       0  
167791  8.41702       8.27349  1.38770       0  
233078  8.67603       8.26190  1.60590       0

primary Board Game Rank  \
id                                                                          
274471                           Malhya: Lands of Legends           13627   
275777                                              Erune           14434   
228370                                        TerroriXico           17555   
345976  System Gateway (fan expansion for Android: Net...            7430   
322354                                           DEFCON 1           11482   

        usersrated  average  bayesaverage    stddev  median  
id                                                           
274471          41  9.56829       5.52555  0.644844       0  
275777          43  9.55814       5.52080  1.244570       0  
228370          70  9.42857       5.50306  1.695130       0  
345976          47  9.40425       5.59984  0.965568       0  
322354          60  9.35867       5.54021  1.675870       0

These columms are interesting, as they provide information on the review scores of the games. BGG ranks games based on the `bayesaverage` column, or 'Geek Rating'. This appears to be done primarily to reduce the chance that games with high avarages and low review scores, such as the top 5 sorted by average, get to the top of the rankings. The way that the bayes average is worked out is unknown, as to avoid people gaming the system.

A low standard deviation in this case would mean that people are consistently rating the game close to the average score.
The median column that I pulled, thinking it would be the median rating, appears to not be the case.

In [33]:
df_game_details['median'].unique()

array([0], dtype=int64)

Since the only value is `median` is `0`, I'll just drop the column, as it doesn't provide any useful information.

In [35]:
cols_to_drop = [
    'Unnamed : 0',
    'type',
    'thumbnail',
    'image',
    'alternate',
    ]